File: cs/io/gfx/Vertex.cs
using VoidEngine.Math;

namespace VoidEngine.IO.GFX
{
	public struct Vertex
	{
		Vec3f Position { get; set; }
		Vec4f Color { get; set; }
		Vec2f UV { get; set; }

		public Vertex(Vec3f position, Vec4f color, Vec2f uv)
		{
			this.Position = position;
			this.Color = color;
			this.UV = uv;
		}
	}
}

File: src/io/gfx/vertex.cpp
#include "ve/struct_db.hpp"
#include "ve/variant.hpp"
#include <ve/io/gfx/vertex.hpp>

namespace VoidEngine::IO::GFX {
	StructData VertexData = {
		.name = "Vertex",
		.members = {
			{ .name = "position",	.type = VariantType::STRUCT, .data = &Vec3Data },
			{ .name = "color",		.type = VariantType::STRUCT, .data = &Vec4Data },
			{ .name = "UV",			.type = VariantType::STRUCT, .data = &Vec2Data },
			{ .name = "normal",		.type = VariantType::STRUCT, .data = &Vec3Data },
		}
	};
}
